presto-spi/src/main/java/com/facebook/presto/spi/PageBuilder.java

```diff
@@ -77,14 +78,24 @@ private PageBuilder(int initialExpectedEntries, int maxPageBytes, List<? extends
             }
         }
 
-        blockBuilders = new BlockBuilder[types.size()];
-        for (int i = 0; i < blockBuilders.length; i++) {
-            blockBuilders[i] = types.get(i).createBlockBuilder(
-                    pageBuilderStatus.createBlockBuilderStatus(),
-                    expectedEntries,
-                    pageBuilderStatus.getMaxBlockSizeInBytes() / expectedEntries);
-        }
         this.initialExpectedEntries = expectedEntries;
```

Comment:
I would move all of the `expectedEntries` calculation inside of the `!templateBlockBuilders.isPresent()` else branch.  I'd remove the `initialExpectedEntries` field